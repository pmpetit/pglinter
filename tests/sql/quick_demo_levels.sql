-- Quick demonstration of the new rule level management functions
-- Run this in PostgreSQL after installing the updated pglinter extension
CREATE EXTENSION pglinter;

\echo '=== Rule Level Management Demo ==='

-- Show current T005 levels (should be 50, 90 from rules.sql)
\echo 'Current T005 levels:'
SELECT pglinter.get_rule_levels('T005') as current_t005_levels;

-- Update T005 to be more strict (lower thresholds)
\echo 'Making T005 more strict (warning=20, error=40):'
SELECT pglinter.update_rule_levels('T005', 20, 40) as update_success;

-- Verify the change
\echo 'T005 levels after update:'
SELECT pglinter.get_rule_levels('T005') as updated_t005_levels;

-- Check a few other rules
\echo 'Current levels for various rules:'
SELECT
    code,
    warning_level,
    error_level,
    pglinter.get_rule_levels(code) as formatted_levels
FROM pglinter.rules
WHERE code IN ('B001', 'T001', 'T005')
ORDER BY code;

-- Update only warning level for B001
\echo 'Updating only B001 warning level to 5:'
SELECT pglinter.update_rule_levels('B001', 5, NULL) as partial_update;

-- Show the result
\echo 'B001 after partial update:'
SELECT pglinter.get_rule_levels('B001') as b001_levels;

\echo '=== Demo Complete ==='
\echo 'New functions available:'
\echo '  - pglinter.get_rule_levels(rule_code)'
\echo '  - pglinter.update_rule_levels(rule_code, warning_level, error_level)'
\echo ''
\echo 'Use NULL for warning_level or error_level to keep current value unchanged.'

DROP EXTENSION pglinter CASCADE;
