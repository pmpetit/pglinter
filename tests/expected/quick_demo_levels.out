-- Quick demonstration of the new rule level management functions
-- Run this in PostgreSQL after installing the updated pglinter extension
CREATE EXTENSION pglinter;
\echo '=== Rule Level Management Demo ==='
=== Rule Level Management Demo ===
-- Show current T005 levels (should be 50, 90 from rules.sql)
\echo 'Current T005 levels:'
Current T005 levels:
SELECT pglinter.get_rule_levels('T005') as current_t005_levels;
      current_t005_levels       
--------------------------------
 warning_level=1, error_level=1
(1 row)

-- Update T005 to be more strict (lower thresholds)
\echo 'Making T005 more strict (warning=20, error=40):'
Making T005 more strict (warning=20, error=40):
SELECT pglinter.update_rule_levels('T005', 20, 40) as update_success;
NOTICE:  ✅ Updated rule T005 levels: warning=20, error=40
 update_success 
----------------
 t
(1 row)

-- Verify the change
\echo 'T005 levels after update:'
T005 levels after update:
SELECT pglinter.get_rule_levels('T005') as updated_t005_levels;
       updated_t005_levels        
----------------------------------
 warning_level=20, error_level=40
(1 row)

-- Check a few other rules
\echo 'Current levels for various rules:'
Current levels for various rules:
SELECT
    code,
    warning_level,
    error_level,
    pglinter.get_rule_levels(code) as formatted_levels
FROM pglinter.rules
WHERE code IN ('B001', 'T001', 'T005')
ORDER BY code;
 code | warning_level | error_level |         formatted_levels         
------+---------------+-------------+----------------------------------
 B001 |            20 |          80 | warning_level=20, error_level=80
 T001 |             1 |           1 | warning_level=1, error_level=1
 T005 |            20 |          40 | warning_level=20, error_level=40
(3 rows)

-- Update only warning level for B001
\echo 'Updating only B001 warning level to 5:'
Updating only B001 warning level to 5:
SELECT pglinter.update_rule_levels('B001', 5, NULL) as partial_update;
NOTICE:  ✅ Updated rule B001 levels: warning=5, error=80
 partial_update 
----------------
 t
(1 row)

-- Show the result
\echo 'B001 after partial update:'
B001 after partial update:
SELECT pglinter.get_rule_levels('B001') as b001_levels;
           b001_levels           
---------------------------------
 warning_level=5, error_level=80
(1 row)

\echo '=== Demo Complete ==='
=== Demo Complete ===
\echo 'New functions available:'
New functions available:
\echo '  - pglinter.get_rule_levels(rule_code)'
  - pglinter.get_rule_levels(rule_code)
\echo '  - pglinter.update_rule_levels(rule_code, warning_level, error_level)'
  - pglinter.update_rule_levels(rule_code, warning_level, error_level)
\echo ''

\echo 'Use NULL for warning_level or error_level to keep current value unchanged.'
Use NULL for warning_level or error_level to keep current value unchanged.
DROP EXTENSION pglinter CASCADE;
